### PR TITLE
feat(esf): improve conditional search accessibility - 16.1.x

### LINKED ISSUE #13522: [Keyboard Navigation>] Keyboard focus is not in logical order under 'Text filter' popup, while navigating using 'tab key.
Keyboard focus is not going in logical order for Product Name header filter. By moving through tab key inside the text filter the focus is not going in logical order.

Shift + Tab Key - going directly to Apply button but require going on text filter button.

source link- [Excel Style Filtering in Angular Data Grid- Ignite UI for Angular (infragistics.com)](https://www.infragistics.com/products/ignite-ui-angular/angular/components/grid/excel-style-filtering)

Repo Step:

1. Launch the above link.
2. Go to Product Name Header Filter.
3. Navigate to the 'text filter' button.
4. Verify whether keyboard focus is not in logical order under 'Text filter' popup or not.

Actual Result:
Keyboard focus is not in logical order under 'Text filter' popup, while navigating using tab key.
Observation: After pressing 'Shit + tab' key on contain menu keyboard focus moves to 'apply' button and again pressing 'shift + tab' its moves 'cancel 'button.
Expected Result:
Keyboard focus should be in logical order under 'Text filter' popup, while navigating using tab key.
Ex: Keyboard focus should move back to 'text filter' control while navigating using 'shift + tab' key.

### LINKED ISSUE #13531: [Keyboard Navigation-Name menu-Text filter] 'Text filter' popup is not getting by 'Esc' key under 'Name' dialog.
User Experience:
The "Esc" key is a common and expected method to close modals, and its absence can hinder their ability to move through the interface. This can interrupt the natural flow of navigation and interaction, making the experience more cumbersome.

'Text filter' popup is not getting closed by 'Esc' key under 'Name' dialog.
Observation: When the keyboard focus lands on 'text filter' control and after invoking it popup appears and 
after pressing 'esc' key it is not getting closed the popup and by pressing shift + tab key it is moving to 'apply' button.

source link: [Excel Style Filtering in Angular Data Grid- Ignite UI for Angular (infragistics.com)](https://www.infragistics.com/products/ignite-ui-angular/angular/components/grid/excel-style-filtering)

### PR DESCRIPTION
**Pull request by onlyexeption**
Closes #13522 
Closes #13531

### Additional information (check all that apply):
 - [ ] Bug fix
 - [x] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/grids/filtering/excel-style/excel-style-conditional-filter.component.html [hunk 1]
```diff
@@ -1,5 +1,6 @@
 <ng-container *ngIf="esf.column">
     <div tabindex="0"
+        id='excelFilter'
```
Comment (igdmdimitrov) [inline projects/igniteui-angular/src/lib/grids/filtering/excel-style/excel-style-conditional-filter.component.html:3]:
```suggestion
        id='esfConditionalFilter'
```

Comment (igdmdimitrov) [reply projects/igniteui-angular/src/lib/grids/filtering/excel-style/excel-style-conditional-filter.component.html:3]: @onlyexeption , do the same rename in `drop-down.component.ts` and `excel-style-conditional-filter.component.ts`

Comment (ChronosSF) [reply projects/igniteui-angular/src/lib/grids/filtering/excel-style/excel-style-conditional-filter.component.html:3]: @onlyexeption , this is also considered a bug and should be fixed for 15.1.x

**Comment by igdmdimitrov** [comment]
PR https://github.com/IgniteUI/igniteui-angular/pull/13591 will address these issues.

**Pull request closed without merging**